### DISCUSSION TITLE
브랜치 전략

### DISCUSSION THREAD

**Discussion by codeisneverodd** [prgrms-fe-devcourse/FEDC2_TimeDivider_Root#1] [1. 비개발]
브랜치의 규칙을 정하는 것이 좋을 것 같습니다! 의견 남겨주세요

**Comment by ljw0096** [comment]
Git-flow는 총 5가지의 브랜치를 사용해서 운영을 합니다. 

- master  : 기준이 되는 브랜치로 제품을 배포하는 브랜치 입니다. 
- develop : 개발 브랜치로 개발자들이 이 브랜치를 기준으로 각자 작업한 기능들을 합(Merge)칩니다. 
- feature : 단위 기능을 개발하는 브랜치로 기능 개발이 완료되면 develop 브랜치에 합칩니다.
- release : 배포를 위해 master 브랜치로 보내기 전에 먼저 QA(품질검사)를 하기위한 브랜치 입니다. 
- hotfix  : master 브랜치로 배포를 했는데 버그가 생겼을 떄 긴급 수정하는 브랜치 입니다.
출처: https://uxgjs.tistory.com/183 [UX 공작소:티스토리]

**Comment by codeisneverodd** [comment]
master -> main
main, develop, release 는 protected브랜치 유지
feature, hotfix  는 prefix로 사용해요! 예를 들어 타이머를 만든다고 하면 feature-timer 이런식으로 해도 좋을 것 같아요. 급한 수정이라면 hotfix-timer 이런식으로요!

**Reply by dahye1013** [reply]
`feature-timer`은 조금 모호한 느낌이 들어요 🤔 
기능 수정 대상이 component인지 page인지 suffix까지 추가하면 어떨까요?
